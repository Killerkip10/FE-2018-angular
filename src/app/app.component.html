<app-header></app-header>

<mat-drawer-container class="container">
  <mat-drawer *ngIf="tokenService.getToken()" mode="side" opened>
    <mat-nav-list>
      <a mat-list-item [routerLink]="['user-profile']" routerLinkActive="active">Profile</a>
      <a mat-list-item [routerLink]="['user-list']" routerLinkActive="active">Users list</a>

    </mat-nav-list>
  </mat-drawer>

  <mat-drawer-content>
    <router-outlet></router-outlet>
  </mat-drawer-content>
</mat-drawer-container>

<app-footer></app-footer>
