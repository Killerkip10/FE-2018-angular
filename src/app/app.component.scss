app-header{
  display: block;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 2;
}
.container {
  margin-top: 56px;
  min-height: calc(100vh - 112px);
  mat-drawer{
    position: fixed;
    top: 56px;
    bottom: inherit;
  }
  .active {
    background-color: rgba(0,0,0,.04);
  }
}
