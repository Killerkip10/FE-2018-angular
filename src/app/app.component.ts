import {Component} from '@angular/core';
import {TranslateService} from '@ngx-translate/core';

import {TokenService} from './core/services';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent {
  private tokenService = TokenService;

  constructor(translateService: TranslateService) {
    translateService.addLangs(['en', 'ru']);
    translateService.setDefaultLang('en');
  }
}
