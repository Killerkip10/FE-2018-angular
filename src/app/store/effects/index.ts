import {ProfileEffects} from './profile';
import {UsersEffects} from './users';

export const effects = [
  ProfileEffects,
  UsersEffects
];
