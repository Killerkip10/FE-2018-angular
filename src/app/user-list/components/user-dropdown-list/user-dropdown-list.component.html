<div class="list">
  <div>
    <mat-form-field>
      <input matInput #searchUser type="text" placeholder="Search">
      <mat-spinner *ngIf="spinner" matSuffix [diameter]="18"></mat-spinner>
    </mat-form-field>
  </div>

  <div>
    <app-user-dropdown-item
      *ngFor="let item of input$ | async"
      [user]="item"
      (click)="select.emit(item)">
    </app-user-dropdown-item>
  </div>
</div>
