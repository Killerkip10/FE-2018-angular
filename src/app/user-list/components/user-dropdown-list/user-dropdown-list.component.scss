.list {
  overflow: hidden;
  mat-form-field{
    width: 100%;
  }
}
