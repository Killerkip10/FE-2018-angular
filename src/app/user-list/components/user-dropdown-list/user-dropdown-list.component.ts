import {Component, ElementRef, OnInit, ViewChild, Output, EventEmitter, AfterViewInit} from '@angular/core';
import {fromEvent, Observable} from 'rxjs';
import {debounceTime, switchMap, tap} from 'rxjs/operators';

import {User} from '../../../models';
import {SearchUserService} from '../../../core/services';

interface InputEvent {
  target: {value: string};
}

@Component({
  selector: 'app-user-dropdown-list',
  templateUrl: './user-dropdown-list.component.html',
  styleUrls: ['./user-dropdown-list.component.scss']
})
export class UserDropdownListComponent implements OnInit, AfterViewInit {
  @ViewChild('searchUser') searchUser: ElementRef;
  @Output() select = new EventEmitter<User>();

  public input$: Observable<User[]>;
  public spinner = false;

  constructor(private searchUserService: SearchUserService) {}

  ngOnInit(): void {
    this.input$ = fromEvent<InputEvent>(this.searchUser.nativeElement, 'input')
      .pipe(
        debounceTime(500),
        tap(() => this.spinner = true),
        switchMap(v => this.searchUserService.findUser(v.target.value)),
        tap(() => this.spinner = false),
      );
   }
  ngAfterViewInit(): void {
    this.searchUser.nativeElement.dispatchEvent(new Event('input'));
  }
}
