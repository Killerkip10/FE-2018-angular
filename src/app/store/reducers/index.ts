import {ActionReducerMap} from '@ngrx/store';

import * as fromUsers from './users';
import * as fromProfile from './profile';

export interface State {
  users: fromUsers.State;
  profile: fromProfile.State;
}

export const reducers: ActionReducerMap<State> = {
  users: fromUsers.reducer,
  profile: fromProfile.reducer
};
