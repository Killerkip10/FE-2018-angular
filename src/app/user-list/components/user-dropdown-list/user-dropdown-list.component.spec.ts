import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import {NO_ERRORS_SCHEMA} from '@angular/core';

import { UserDropdownListComponent } from './user-dropdown-list.component';
import {TranslateModule} from '@ngx-translate/core';
import {StoreModule} from '@ngrx/store';
import {reducers} from '../../../store/reducers';
import {EffectsModule} from '@ngrx/effects';
import {effects} from '../../../store/effects';
import {CoreModule} from '../../../core/core.module';
import {AppTranslate} from '../../../app.translate';

describe('ItemComponent', () => {
  let component: UserDropdownListComponent;
  let fixture: ComponentFixture<UserDropdownListComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        CoreModule,
        TranslateModule,
        StoreModule.forRoot(reducers),
        EffectsModule.forRoot(effects),
        AppTranslate
      ],
      declarations: [
        UserDropdownListComponent
      ],
      schemas: [
        NO_ERRORS_SCHEMA
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(UserDropdownListComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
