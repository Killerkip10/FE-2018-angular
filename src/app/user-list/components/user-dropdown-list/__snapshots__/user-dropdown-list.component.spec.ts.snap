// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDropDownListComponent snapshot 1`] = `
<app-user-dropdown-list
  input$={[Function Subscriber]}
  searchUser={[Function ElementRef]}
  select={[Function EventEmitter]}
  store={[Function Store]}
>
  <div
    class="list"
  >
    <div>
      <mat-form-field>
        <input
          matinput=""
          placeholder="SEARCH"
          type="text"
        />
      </mat-form-field>
    </div>
    <div>
      
    </div>
  </div>
</app-user-dropdown-list>
`;

exports[`UserDropDownListComponent snapshot 2`] = `
<app-user-dropdown-list
  input$={[Function Subscriber]}
  searchUser={[Function ElementRef]}
  select={[Function EventEmitter]}
  store={[Function Store]}
  users={[Function Array]}
>
  <div
    class="list"
  >
    <div>
      <mat-form-field>
        <input
          matinput=""
          placeholder="SEARCH"
          type="text"
        />
      </mat-form-field>
    </div>
    <div>
      
    </div>
  </div>
</app-user-dropdown-list>
`;

exports[`UserDropDownListComponent snapshot 3`] = `
<app-user-dropdown-list
  input$={[Function Subscriber]}
  searchUser={[Function ElementRef]}
  select={[Function EventEmitter]}
  store={[Function Store]}
  users={[Function Array]}
>
  <div
    class="list"
  >
    <div>
      <mat-form-field>
        <input
          matinput=""
          placeholder="SEARCH"
          type="text"
        />
      </mat-form-field>
    </div>
    <div>
      
      <app-user-dropdown-item />
      <app-user-dropdown-item />
    </div>
  </div>
</app-user-dropdown-list>
`;
